src/interface/contract.rs

```diff
@@ -84,6 +82,10 @@ impl KnownState for AllocatedState {}
 pub type OwnedAllocation = OutputAssignment<AllocatedState>;
 pub type RightsAllocation = OutputAssignment<VoidState>;
 pub type FungibleAllocation = OutputAssignment<Amount>;
+pub type InflationAllowanceAllocation = OutputAssignment<Amount>;
+pub type UpdateRights = OutputAssignment<VoidState>;
+pub type BurnEpoch = OutputAssignment<VoidState>;
+pub type BurnRight = OutputAssignment<VoidState>;
```

Comment:
I think we do not need these newtypes (and certainly not in the global namespace, since they are RGB20-specific).

Just use `FungibleAllocation`, and `RightsAllocation` instead in the code you have added. The rest looks good

Reply:
The other PR is merged, so can you pls update this PR with removing these type names - and I will merge it and proceed to the next ones

Reply:
@cymqqqq your PRs share the same commits, thus if you update one of them the other will have a diverging history. I suggest to split the commits into PRs (such that each one has one or two) - or just give the one PR with all commits. Otherwise the process will be quite complex

Reply:
I have updated this PR, and thanks, I will try my best to just give one PR later. :)